core/drand_control.go

```diff
@@ -292,14 +292,17 @@ func (d *Drand) setupAutomaticDKG(c context.Context, in *drand.InitDKGPacket) (*
 	}
 
 	d.log.Debug("init_dkg", "send_key", "leader", lpeer.Address())
-	err = d.privGateway.ProtocolClient.SignalDKGParticipant(context.Background(), lpeer, prep)
+	nc, cancel := context.WithTimeout(context.Background(), MaxWaitPrepareDKG)
```

Comment:
not `context.WithTimeout(c, MaxWaitPrepareDKG)` ?

Reply:
If I'm taking `c` that means I am using the gRPC provided context that will cancels out when the CLI ctrl-c'.

Reply:
seems fine. lint says you should change `c context.Context` in the argument to the function to `_ context.Context` since the parameter is unused now.